subsys/settings/Kconfig

```diff
@@ -24,6 +24,20 @@ config SETTINGS_RUNTIME
 	help
 	  Enables runtime storage back-end.
 
+config SETTINGS_ROM_HANDLERS
```

Comment:
I expected that static ones are always available to use otherwise we might have to make macros to fail at build time if the config setting is not selected which imo has no real benefit.

Also we these 2 options one can actually enable the settings without enabling either types of handles which would turn the settings useless since no handlers can be registered.

Reply:
I tried to get the code "compatible" with the present case. If you now set SETTINGS_ROM_HANDLERS to n you should get the exact same size as we get in the present code. On the otherside to generate the smallest RAM usage you can set SETTINGS_RAM_HANDLERS to n.

When they are both set to n you will not be able to call SETTINGS_ROM_REGISTER() or settings_register() so compilation would fail.

I agree that it might lead to confusion, if we can agree on always enabling the ROM (static) handling, I will remove the option SETTINGS_ROM_HANDLERS.

Reply:
Id go for having static support allways when CONFIG_SETTINGS is set, this makes it a lot simpler in terms of config options and I assume the vast majority of the handlers can be tranformed in static handlers quite easily, to be honest I think the dynamic handlers might only be useful for application or to override but you said that is not even possible right now.

Reply:
OK, I will by default enable the static handlers.